File: python/examples/05_ecosystem_webhooks.py
import json
import hashlib
import hmac
from uiid_client import UIIDClient

def main():
    print("=== 05. Ecosystem, Auditing & Webhooks Workflow (Python) ===\n")

    client = UIIDClient()

    # Webhook signature verification demo
    payload = {"event": "alias.data.updated", "alias_id": "UIID-Alias-80BB"}
    secret = "my_webhook_secret_123"
    body_bytes = json.dumps(payload, separators=(',', ':')).encode('utf-8')
    sig = hmac.new(secret.encode('utf-8'), body_bytes, hashlib.sha256).hexdigest()

    is_valid = UIIDClient.verify_webhook_signature(payload, sig, secret)
    print(f"[Webhooks] HMAC-SHA256 Signature Verification: {is_valid}\n")

    print("[Ecosystem & Audit] Methods available in Python SDK:")
    print(" - client.get_marketplace_apps()")
    print(" - client.get_user_badges()")
    print(" - client.get_core_audit_logs()")
    print(" - client.get_alias_audit_logs(alias_id='UIID-Alias-123')\n")

if __name__ == "__main__":
    main()

File: php/examples/05_ecosystem_webhooks.php
<?php

require_once __DIR__ . '/../vendor/autoload.php';

use UIID\SDK\UIIDClient;

echo "=== 05. Ecosystem, Auditing & Webhooks Workflow (PHP) ===\n\n";

$payload = ['event' => 'alias.data.updated', 'alias_id' => 'UIID-Alias-80BB'];
$secret = 'whsec_secret_123';
$bodyStr = json_encode($payload, JSON_UNESCAPED_SLASHES);
$signature = hash_hmac('sha256', $bodyStr, $secret);

$isValid = UIIDClient::verifyWebhookSignature($payload, $signature, $secret);
echo "[Webhooks] HMAC Verification: " . ($isValid ? 'PASSED' : 'FAILED') . "\n\n";

echo "[Ecosystem & Audit] Available PHP SDK Methods:\n";
echo " - \$client->getMarketplaceApps()\n";
echo " - \$client->getUserBadges()\n";
echo " - \$client->getCoreAuditLogs()\n";
echo " - \$client->getAliasAuditLogs('UIID-Alias-80BB')\n\n";
